backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantController.java

```diff
@@ -15,12 +17,33 @@ public class PlantController {
     private final PlantService plantService;
 
     @PostMapping
-    public Plant createNewPlant (@RequestBody Plant plant, Principal principal){
-        return plantService.createPlant(plant, principal);
+    public Plant createNewPlant(@RequestBody Plant plant, Principal principal) {
+        return plantService.createPlant(plant, principal.getName());
     }
 
     @GetMapping
-    public Collection<Plant> getAllPlants(Principal principal){
-        return plantService.getAllPlantsByUser(principal);
+    public Collection<Plant> getAllPlants(Principal principal) {
+        return plantService.getAllPlantsByUser(principal.getName());
     }
+
+    @GetMapping("/{id}")
+    public ResponseEntity <Plant> getPlantById(@PathVariable String id, Principal principal){
+        return ResponseEntity.of(plantService.getPlantByIdAndUser(id, principal.getName()));
+    }
+
+    @GetMapping("{/searchedPlant}")
```

Comment:
That won't work, it needs to be like this: `"/{searchedPlant}"`

Reply:
Sorry, Typo Phd at work!